.github/workflows/main.yml

```diff
@@ -0,0 +1,394 @@
+        uses: actions/cache@v2
+        with:
+          path: /tmp/luarocks
+          # The build input for luarocks changes per test, so we need separate caches
+          key: ${{ github.workflow }}-${{ runner.os }}-${{ matrix.test_name }}-luarocks-3.5.0
+
+      - name: Install fresh Luarocks
+        if: steps.cache-luarocks.outputs.cache-hit != 'true'
+        run: |
+          wget -O /tmp/luarocks.tar.gz https://github.com/luarocks/luarocks/archive/v3.5.0.tar.gz
+          mkdir /tmp/luarocks
+          tar -xf /tmp/luarocks.tar.gz -C /tmp/luarocks --strip-components=1
+          cd /tmp/luarocks
+          ./configure --lua-version=${{ matrix.lua_version }} --with-lua-include=${LUAINCLUDE} ${{ matrix.luarocks_args }}
+          make build
+          sudo make install
+
+      - name: Install cached Luarocks
+        if: steps.cache-luarocks.outputs.cache-hit == 'true'
+        run: |
+          cd /tmp/luarocks
+          sudo make install
+
+      - name: Cache xcb-errors
+        id: cache-xcb-errors
+        uses: actions/cache@v2
+        with:
+          path: /tmp/xcb-errors
+          # xcb-errors doesn't have versioned releases, so we checkout and cache by commit.
+          # Make sure to keep this in sync with the `git checkout` below.
+          key: ${{ github.workflow }}-${{ runner.os }}-xcb-errors-${{ env.XCB_ERRORS_SHA }}
+
+      - name: Install fresh xcb-errors
+        if: steps.cache-xcb-errors.outputs.cache-hit != 'true'
+        run: |
+          git clone --recursive https://gitlab.freedesktop.org/xorg/lib/libxcb-errors.git /tmp/xcb-errors
+          cd /tmp/xcb-errors
+          git checkout ${{ env.XCB_ERRORS_SHA }}
+          ./autogen.sh --prefix=/usr
+          make
+          sudo make install
+
+      - name: Install cached xcb-errors
+        if: steps.cache-xcb-errors.outputs.cache-hit == 'true'
+        run: |
+          cd /tmp/xcb-errors
+          sudo make install
+
+      - name: Install rocks
+        run: |
+          sudo -H luarocks install lgi ${{ matrix.lgi_version }}
+          sudo -H luarocks install ldoc
+          sudo -H luarocks install busted
+
+      - name: Install QA check rocks
+        if: matrix.check_qa
+        run: |
+          sudo -H luarocks install luacheck
+          sudo -H luarocks install depgraph
+
+      - name: Install cluacov rock
+        if: matrix.coverage
+        run: sudo -H luarocks install cluacov
+
+      - name: Install coveralls rock
+        if: matrix.coverage == 'coveralls'
+        run: sudo -H luarocks install luacov-coveralls
+
+      - name: Install codecov.io uploader
+        if: matrix.coverage == 'codecov'
+        run: wget -O /tmp/codecov-bash https://codecov.io/bash
+
+      # Check out repository to ${{ github.workspace }}
+      # Automatically picks the current branch/PR
+      - uses: actions/checkout@v2
+
+      - name: Create Build Environment
+        run: cmake -E make_directory -B "${{ github.workspace }}/build"
```

Comment:
Nothing should be wrong with that, as far as I know.
This is simply one of the few lines from GitHub's CMake template that I never had to touch since I first started all this.